cantools/database/can/formats/sym.py

```diff
@@ -921,7 +921,10 @@ def _dump_message(message: Message, signals: List[Signal], min_frame_id: TypingO
         if multiplexer_signal.byte_order == 'big_endian':
             m_flag = '-m'
         hex_multiplexer_id = format(multiplexer_id, 'x').upper()
-        message_str += f'Mux="{hex_multiplexer_id}" {_convert_start(multiplexer_signal.start, multiplexer_signal.byte_order)},{multiplexer_signal.length} {hex_multiplexer_id}h {m_flag}\n'
+        multiplexer_signal_name = multiplexer_signal.name
+        if not multiplexer_signal_name:
```

Comment:
hm, come think about it, I think that all signals always have to have names, so this `if` statement can probably be removed entirely. I just wonder why line 926 seems to be covered by the unit tests...

Reply:
That's right. When I create a symbol and want to set the name of the multiplexer empty, I cannot do that because it's invalid, so all signals always have to have names. However, if we rename the name through python code, like [here](https://github.com/cantools/cantools/pull/579/files#diff-82fbc96aa3f082d4667c09a2a35aec050c73120de9174e1f37bef26ef9cd3115R3415), it works and we can rename the multiplexer to be empty. In that case, the name of the multiplexer would be a empty string.

I think [this suggestion](https://github.com/cantools/cantools/pull/579#issuecomment-1587986038) is great, so I update the test case to cover this case. Refer to [the change](https://github.com/cantools/cantools/pull/579/files#diff-82fbc96aa3f082d4667c09a2a35aec050c73120de9174e1f37bef26ef9cd3115R3412) for function `test_multiplex_sym_dump`.

Reply:
ok. I think that we should not support corrupt databases. IOW,  `multiplexer_signal.name` should be assumed to be a non-empty string. once this is cleaned up, this PR can be merged IMO...

Reply:
You're right. We should not support that. I would prefer the program raises an error to inform that.